### ISSUE TITLE
Write Application Telemetry to JSON file

### ISSUE THREAD

**Issue by kwandrew2** [pyrevitlabs/pyRevit#2144]
## Is your feature request related to a problem? Please describe

It may be possible and I'm just missing a setting but we currently write our script telemetry data to JSON files and would like to do the same with Application Telemetry.

## Describe the solution you'd like

I'd like to save application telemetry data to JSON files.  We use power BI to visualize pyRevit tool usage (from the JSON files) and would like to use the same process for application telemetry.

## Describe alternatives you've considered

I requested our IT team setup a application telemetry server or API endpoint and they made this process seem prohibitive because of the effort/time investment it'd take. 

## Additional context

If writing application telemetry to JSON is already possible please let me know and I can put this to rest.  If it's not possible to write Application Telemetry data to JSON files, adding a note next to those sections in the pyRevit settings could potentially save others on these efforts.

Thanks in advance!

**Comment by sanzoghenzo** [comment]
Hi @kwandrew2 ,
 have you tried this?
![image](https://github.com/eirannejad/pyRevit/assets/977953/37953b85-28f6-4749-95a7-15cfb1905923)

**Comment by kwandrew2** [comment]
Yes but it doesn’t seem to write hook/event data to the json, only commands.

**Comment by sanzoghenzo** [comment]
Thanks for the clarification, I admin I don't use telemetry (yet) and I didn't know the difference between script and application telemetry 😅 

To write the application telemetry to the same json file it would be enough to copy over 3 lines of code from [here](https://github.com/eirannejad/pyRevit/blob/55ad8a3780c23eb832bfa65fe88878afad4d42ec/pyrevitlib/pyrevit/runtime/ScriptTelemetry.cs#L97C1-L100C92) to [here](https://github.com/eirannejad/pyRevit/blob/55ad8a3780c23eb832bfa65fe88878afad4d42ec/pyrevitlib/pyrevit/runtime/EventTelemetry.cs#L271).

You can already try it on you local copy (files are in `%appdata%\pyRevit-Master\` if you installed it as a single user), but I'm not sure if there were any reason to left it out; tomorrow at the online meeting we could discuss it along with my proposal: to get rid of our custom telemetry server and use OpenTelemetry, so that one can use the server and visualization app of his/her choice (Jaeger and zipkin are two open source options, among others - you IT would just need a linux box with docker to spin up a server...)

**Comment by kwandrew2** [comment]
Thank you for the assistance!  I ended up playing around with hooks a bit and enlisted the help of one of my coworkers to get project times written to a file.  I’m waiting on the end product yet but truly appreciate the assistance and may end up giving what you sent a shot if we run into any issues.

**Comment by MohamedAsli** [comment]
@kwandrew2, can you tell us what you did? This could help others facing the same issue.

From my side, I use JSON to collect logs and Power BI for visualization. I manage to catch commands and hooks without the need to set up a server.
But you’re right about the **built-in** Application Hooks: they require a telemetry server and don’t use the JSON output format. Still, nothing prevents you from creating equivalent hooks in a .py format to catch them anyway, and they’ll be written to your JSON log file.

I’ll close this issue as I consider the problem solved, but if you’d like to dig deeper into the topic, feel free to reopen it.
Happy to help anytime :)

**Comment by kwandrew2** [comment]
Someone on our team was able to write a hook which captured the project opening, opened, syncing, synced times and writes those to an excel file which we use powerBi to view.  So I believe our resolution was just as you stated!
